pint/facets/numpy/numpy_func.py

```diff
@@ -726,6 +726,36 @@ def _prod(a, *args, **kwargs):
     implement_prod_func(name)
 
 
+def _base_unit_if_needed(a):
+    if a._is_multiplicative:
+        return a
+    else:
+        if a.units._REGISTRY.autoconvert_offset_to_baseunit:
+            return a.to_base_units()
+        else:
+            raise OffsetUnitCalculusError(a.units)
+
+
+@implements("trapz", "function")
+def _trapz(a, x=None, dx=1.0, **kwargs):
+    a = _base_unit_if_needed(a)
+    if x is not None:
+        x = _base_unit_if_needed(x)
+        units = a.units
+        if hasattr(x, "units"):
+            units *= x.units
+            x = x._magnitude
+        ret = np.trapz(a._magnitude, x, **kwargs)
+    else:
+        dx = _base_unit_if_needed(dx)
+        if hasattr(dx, "units"):
+            units *= dx.units
```

Comment:
This is failing as `units` is not defined.

Reply:
Fixed.